torch/csrc/autograd/python_engine.cpp

```diff
@@ -144,6 +144,10 @@ PyObject *THPEngine_run_backward(THPEngine *self, PyObject *args, PyObject *kwar
     THPUtils_assert(THPVariable_Check(_tensor), "element %d of tensors "
         "tuple is not a Tensor", i);
     auto& variable = ((THPVariable*)_tensor)->cdata;
+    if(variable.is_complex()) {
```

Comment:
Given that the code above allow complex backward for both the python and the cpp api. Maybe you want to add the same thing to the regular engine run_backward function.